custom_components/adaptive_lighting/switch.py

```diff
@@ -343,33 +429,103 @@ async def async_setup_entry(
+                        force=True,
+                        context=this_switch.create_context(
+                            "service", parent=service_call.context
+                        ),
+                    )
+
+    @callback
+    async def handle_set_manual_control(service_call: ServiceCall):
+        """Set or unset lights as 'manually controlled'."""
+        _LOGGER.debug(
+            "Called 'adaptive_lighting.set_manual_control' service with '%s'",
+            service_call.data,
+        )
+        these_switches = data = None
+        these_switches, data = _parse_service_args(hass, service_call)
+        lights = data[CONF_LIGHTS]
+        for this_switch in these_switches:
+            if not lights:
+                all_lights = this_switch._lights  # pylint: disable=protected-access
+            else:
+                all_lights = _expand_light_groups(this_switch.hass, lights)
+            if service_call.data[CONF_MANUAL_CONTROL]:
+                for light in all_lights:
+                    this_switch.turn_on_off_listener.manual_control[light] = True
+                    _fire_manual_control_event(this_switch, light, service_call.context)
+            else:
+                this_switch.turn_on_off_listener.reset(*all_lights)
+                # pylint: disable=protected-access
+                if this_switch.is_on:
+                    await this_switch._update_attrs_and_maybe_adapt_lights(
+                        all_lights,
+                        transition=this_switch._initial_transition,
+                        force=True,
+                        context=this_switch.create_context(
+                            "service", parent=service_call.context
+                        ),
+                    )
+
     # Register `apply` service
-    platform = entity_platform.current_platform.get()
-    platform.async_register_entity_service(
-        SERVICE_APPLY,
-        {
-            vol.Optional(
-                CONF_LIGHTS, default=[]
-            ): cv.entity_ids,  # pylint: disable=protected-access
-            vol.Optional(
-                CONF_TRANSITION,
-                default=switch._initial_transition,  # pylint: disable=protected-access
-            ): VALID_TRANSITION,
-            vol.Optional(ATTR_ADAPT_BRIGHTNESS, default=True): cv.boolean,
-            vol.Optional(ATTR_ADAPT_COLOR, default=True): cv.boolean,
-            vol.Optional(CONF_PREFER_RGB_COLOR, default=False): cv.boolean,
-            vol.Optional(CONF_TURN_ON_LIGHTS, default=False): cv.boolean,
-        },
-        handle_apply,
+    hass.services.async_register(
+        domain=DOMAIN,
+        service=SERVICE_APPLY,
+        service_func=handle_apply,
+        schema=vol.Schema(
+            {
+                vol.Optional("entity_id"): cv.entity_ids,
+                vol.Optional(
+                    CONF_LIGHTS, default=[]
+                ): cv.entity_ids,  # pylint: disable=protected-access
```

Comment:
Note to self: check whether pylint comment is needed

Reply:
It's not, the only time I ever needed it was when I worked on the `change_sunlight_settings` PR which modified the frozen SunLightSettings instance directly.